src/legacy/server/index_patterns/mixin.ts

```diff
@@ -55,12 +40,18 @@ export function indexPatternsMixin(kbnServer, server) {
    *  @method request.getIndexPatternsService
    *  @type {IndexPatternsService}
    */
-  server.addMemoizedFactoryToRequest('getIndexPatternsService', request => {
-    const { callWithRequest } = request.server.plugins.elasticsearch.getCluster('data');
-    const callCluster = (...args) => callWithRequest(request, ...args);
-    return server.indexPatternsServiceFactory({ callCluster });
-  });
+  (server as any).addMemoizedFactoryToRequest(
```

Comment:
can you add

```
    addMemoizedFactoryToRequest: (
      name: string,
      factoryFn: (request: Request) => Record<string, any>
    ) => void;
```

to `kbn_server.d.ts` so we don't need to cast as `any` here?

Though if you do that you'll have to edit below to ` (callWithRequest as any)(request, ...args);`.  Can't think of a better way to do that without possibly changing the logic.

Reply:
Actually, your next review gave me a hint to remove `any` next to `callWithRequest`. I've added `APICaller` next to `callCluster` and added argument names.